nemo_aligner/algorithms/critic_server_trainer.py

```diff
@@ -101,18 +115,34 @@ def server_infer(self, **inputs: np.ndarray) -> Dict[str, np.ndarray]:
         choice = ServerSignal.FORWARD.cuda()
         torch.distributed.broadcast(choice, 0)
 
-        rewards, values, exceeded = self.run_inference(inputs=inputs)
+        inputs, extra, prepad_sequence_length = process_inference_request(
+            inputs,
+            pad_to_multiple=self.pad_batch_to_multiple,
+            tokenize_func=self.tokenize_func,
+            strip_sequence_length_to_multiple=self.strip_sequence_length_to_multiple,
+        )
+        rewards, values = self.run_inference(inputs=inputs)
+
+        # if the inference request has extra padding that it doesn't need
+        # then we will pad it back up to the expected padding when returning the values
+        if prepad_sequence_length > values.shape[1]:
+            values = np.pad(
+                values, ((0, 0), (0, prepad_sequence_length - values.shape[1])), mode="constant", constant_values=0
+            )
+        else:
+            values = values[:, :prepad_sequence_length]
```

Comment:
Shouldn't `values.size(1)` be already equal to `prepad_sequence_length` in the `else:` block?

Reply:
ah that's right, fixed. c98b5a04f3752d0e609219f82ea59828e525b909 there is only logic to strip the sequence length so the else block will never be hit